### PR TITLE
Added Z up to Y up conversion as an option

### LINKED ISSUE #21: Add export option for switching Z up to Y up
It will match the FBX, OBJ exporter etc. in Rhino since it's possible someone may be using this transfer data to another Z up application.

Also, the conversion should be done with a transform matrix. Then we can apply the transform to the mesh and copy the float arrays. It should be faster and the code a bit more terse.

### PR DESCRIPTION
**Pull request by jrz371**
Resolves #21

Hopefully a bit easier to go through than the last PR. It cleans up writing the vertex, normal, and texture coordinates buffers quite a bit.

### CODE REVIEW

File: glTF-BinExporter/RhinoDocGltfConverter.cs [hunk 1]
```diff
@@ -107,20 +130,13 @@ private void AddRhinoObjectDraco(Rhino.Geometry.Mesh[] rhinoMeshes, Rhino.DocObj
...
-
             // For each rhino mesh, create gl-buffers, gl-meshes, etc.
             foreach (var rhinoMesh in rhinoMeshes)
             {
-                rhinoMesh.Transform(transformYUp);
+                if(options.MapRhinoZToGltfY)
+                {
+                    rhinoMesh.Transform(ZtoYUp);
+                }
                 rhinoMesh.TextureCoordinates.ReverseTextureCoordinates(1);
```
Comment (Doerge) [inline glTF-BinExporter/RhinoDocGltfConverter.cs:140]: Should the texture coordinate remapping happen, if we don't transform z to y up?

**Review by Doerge — COMMENTED** [review]
Nice! Definitely cleans up the code in some much needed areas!

Do we know any applications that we can test the exported result with? Also considering the UV remapping.

As far as I can see the default value is `false`, is that correct? I think we should aim for defaults that work with _semi-official_ gltf viewers, i.e. it should be `on` by default.

File: glTF-BinExporter/RhinoDocGltfConverter.cs (hunk 1 above)
Comment (jrz371) [reply glTF-BinExporter/RhinoDocGltfConverter.cs:140]: I'm quite sure. Here's what happens if I don't.

image

When a mesh is transformed the texture coordinates themselves aren't changed. The texture coordinates are 2d points that correspond to locations on a texture. 

Here's the model: [Guitar.zip](https://github.com/Stykka/glTF-Bin/files/5983238/Guitar.zip)

**Comment by jrz371** [comment]
If you don't use the draco extensions then Blender has glTF import. I'll add a commit making the default on.

**Pull request merged**